src/util/CompositeAsyncHandler.ts

```diff
@@ -86,8 +87,35 @@ export class CompositeAsyncHandler<TIn, TOut> implements AsyncHandler<TIn, TOut>
       }
     }
 
+    let differentCodes = false;
+    const last = { statusCode: 0, name: '' };
+
+    for (const error of errors) {
+      if (error instanceof HttpError) {
+        if (last.statusCode !== error.statusCode) {
+          if (last.statusCode > 0) {
+            // Don't break yet since there might still be a 500 error in there
+            differentCodes = true;
+          } else {
+            last.statusCode = error.statusCode;
+            last.name = error.name;
+          }
+        }
+      } else {
+        last.statusCode = 500;
+        break;
+      }
+    }
+
     const joined = errors.map((error: Error): string => error.message).join(', ');
+    const message = `No handler supports the given input: [${joined}].`;
 
-    throw new UnsupportedHttpError(`No handler supports the given input: [${joined}].`);
+    if (last.statusCode === 500) {
+      throw new Error(message);
```

Comment:
Why not HTTP? This components does not know what happens down the line.

Reply:
Could do that, would not make a difference though since normal Errors are interpreted as 500 errors at the end (to make sure we always catch all errors that occur).

Reply:
See remark above, we might want to do that more explicitly. Also pertains to #6.